tests/unit/stats.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <limits>
+
+#include <util/meta.hpp>
+
+/* For randomness, distribution testing we need some
+ * statistical tests... */
+
+namespace testing {
+
+// Simple summary statistics (min, max, mean, variance)
+// computed with on-line algorithm.
+//
+// Note that variance is population variance (i.e. with
+// denominator n, not n-1).
+
+struct summary_stats {
+    int n = 0;
+    double max = -std::numeric_limits<double>::infinity();
+    double min = std::numeric_limits<double>::infinity();
+    double mean = 0;
+    double variance = 0;
+};
+
+template <typename Seq>
+summary_stats summarize(const Seq& xs) {
+    summary_stats s;
+
+    for (auto x: xs) {
+        s.min = x<s.min? x: s.min;
+        s.max = x>s.max? x: s.max;
+        double d1 = x-s.mean;
+        s.mean += d1/++s.n;
+        double d2 = x-s.mean;
+        s.variance += d1*d2;
+    }
+
+    if (s.n) {
+        s.variance /= s.n;
+    }
+    return s;
```

Comment:
An empty set will have different `min` and `max` values. Does this make sense? I have no opinion about what is "correct" in this case, but the behavior should be stated clearly in a comment.

Also, does it make sense to define the min and max of a sequence of `int` or `float` as doubles?

Reply:
Regarding empty sequences, none of the fields save `n` are well defined — there should be no expectation that these fields have any meaning in that circumstance.

For `int` and `float`, which can be exactly represented as `double`, it does make sense. For larger integral types (our extended precision floats) one could indeed have a template summary class, but I believe this is overkill for the test code we have.